File: articles/what-the-meta-analyses-say-about-grit/go/fisher_z_demo.go
package main

import (
	"fmt"
	"math"
)

func FisherZ(r float64) float64 {
	return 0.5 * math.Log((1.0+r)/(1.0-r))
}

func InverseFisherZ(z float64) float64 {
	e := math.Exp(2.0 * z)
	return (e - 1.0) / (e + 1.0)
}

func main() {
	r := 0.22
	z := FisherZ(r)
	fmt.Printf("r: %.3f\n", r)
	fmt.Printf("Fisher z: %.3f\n", z)
	fmt.Printf("Back-transformed r: %.3f\n", InverseFisherZ(z))
}

File: articles/what-the-meta-analyses-say-about-grit/rust/fisher_z_demo.rs
fn fisher_z(r: f64) -> f64 {
    0.5 * ((1.0 + r) / (1.0 - r)).ln()
}

fn inverse_fisher_z(z: f64) -> f64 {
    let e = (2.0 * z).exp();
    (e - 1.0) / (e + 1.0)
}

fn main() {
    let r = 0.22;
    let z = fisher_z(r);
    println!("r: {:.3}", r);
    println!("Fisher z: {:.3}", z);
    println!("Back-transformed r: {:.3}", inverse_fisher_z(z));
}
